docs/guides/database.rst

```diff
@@ -193,9 +215,21 @@ $type
 $subtype
 	Entity subtype, e.g. ``'blog'`` or ``'page'``.
 
-By convention entities that have an uploaded avatar or icon will have the ``icontime`` property
-assigned. This means that you can use ``$entity->icontime`` to check if an icon exists for the given
-entity.
+Icons are assigned a timestamp when they're created. You can use ``$object->getIconTs($size)`` to 
+get the timestamp if you need to track changes to the icon.
+
+To check if an icon is set, use ``$object->hasIcon($size)``.
+
+Icon methods support passing an icon type if an entity has more than one icon. For example, a user
+might have an avatar and a cover photo icon. You would pass ``'cover_photo'`` as the icon type:
+
+.. code:: php
+	$object->setIconFromUploadedFile('uploaded_photo', 'cover_photo');
+
+	$object->getIconUrl([
+		'size' => 'medium',
+		'type' => 'cover_photo'
```

Comment:
Having these is fine:
 - medium.jpg
 - large.jpg
 - master.jpg

but it becomes messy when I need to add new image sizes:
 - very_large.jpg
 - very_large_horizontal.jpg
 - even_larger.jpg
 - even_larger_but_square.jpg
 - this_will_seriously_be_the_largest_final_size.jpg
 - oops_i_was_wrong_needed_yet_another_mega_sized.jpg

The example may be slightly exaggerated. :)

Good point about theming, though. I hadn't thought of that.